.github/workflows/build-and-publish.yml

```diff
@@ -13,6 +13,15 @@ jobs:
         uses: actions/setup-java@v1
         with:
           java-version: 11
+          distribution: 'adopt'
+      - name: Cache Maven packages
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+      - name: Run unit tests
+        run: mvn clean test
```

Comment:
The deploy action already runs the unit tests, so we shouldn't need to run them first.